src/MSBuild.UnitTests/XMake_Tests.cs

```diff
@@ -793,6 +793,22 @@ public void GetPropertyWithInvalidProjectThrowsInvalidProjectFileExceptionNotInt
             results.ShouldNotContain(ResourceUtilities.GetResourceString("BuildFailedWithPropertiesItemsOrTargetResultsRequested"));
         }
 
+        [Fact]
+        public void BuildFailsWithBadPropertyName()
+        {
+            using TestEnvironment env = TestEnvironment.Create();
+            TransientTestFile project = env.CreateFile("testProject.csproj", @"
+<Project>
+  <Target Name=""Build"">
+  </Target>
+</Project>
+");
+            string results = RunnerUtilities.ExecMSBuild($" {project.Path} /p:someProperty:fdalse= ", out bool success);
+            success.ShouldBeFalse(results);
+
+            results.ShouldContain("error MSB4177: Invalid property.");
```

Comment:
This error message is localized, so to make it not fail on machines with non-English languages as the default, I'd suggest just checking for MSB4177.